### PR TITLE
feat: add wandb logger

### PR DESCRIPTION
**Pull request by tadejsv**
This adds support for logging with WandB - when we add learning rate scheduling (and proper evaluation) this will be expanded.

Here's how to use it

```python
import finetuner
from finetuner.tuner.callback import WandBLogger

finetuner.fit(
    embed_model,
    callbacks=[WandBLogger()],
    train_data=generate_fashion(num_total=2000),
    eval_data=generate_fashion(is_testset=True, num_total=1000),
)
```

Here's how the logs then look like in wandb

image

### CODE REVIEW

File: finetuner/tuner/callback/wandb_logger.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            step=self._train_step,
+        )
+        self._val_losses = []
+
+    def on_fit_end(self, tuner: 'BaseTuner'):
+        """
+        Called at the end of the ``fit`` method call, after finishing all the epochs.
+        """
+
+        self.wandb_logger.finish()
```
Comment (JoanFM) [inline finetuner/tuner/callback/wandb_logger.py:71]: what does finish do? If it does not properly happen, and any exceptio happen, in what state does it end?

**Review by bwanglzu — APPROVED** [review]
Generally looks good to me, to me it's even better if we call it `WandBCallback` it's easier

File: finetuner/tuner/callback/wandb_logger.py (hunk 1 above)
Comment (tadejsv) [reply finetuner/tuner/callback/wandb_logger.py:71]: https://docs.wandb.ai/ref/python/finish

To tell wandb to finish uploading all data

Comment (JoanFM) [reply finetuner/tuner/callback/wandb_logger.py:71]: then we are not properly passing the exit code or reporting failures
